chapters/annotations.tex

```diff
@@ -383,20 +383,25 @@ \section{Annotations for Simulations}\label{annotations-for-simulations}
 \end{center}
 
 \begin{annotationdefinition}[experiment]
-% henrikt-ma 2021-04: Seems strange to allow 'experiment' completely without list of options -- what would it mean?
 \begin{synopsis}[grammar]\begin{lstlisting}
-"experiment"
-   [ "(" [ experimentOption { "," experimentOption } ] ")" ]
-
-experimentOption:
-   "StartTime" "=" [ "+" | "-" ] UNSIGNED-NUMBER
-   | "StopTime" "=" [ "+" | "-" ] UNSIGNED-NUMBER
-   | "Interval" "=" UNSIGNED-NUMBER
-   | "Tolerance" "=" UNSIGNED-NUMBER
+annotation "experiment" "(" experimentModifiers ")"
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-The \lstinline{experiment} annotation defines the default start time (\lstinline!StartTime!) in {[}s{]}, the default stop time (\lstinline!StopTime!) in {[}s{]}, the suitable time resolution for the result grid (\lstinline!Interval!) in {[}s{]}, and the default relative integration tolerance (\lstinline!Tolerance!) for simulation experiments to be carried out with the model or block at hand.
+The modifiers are for the following record.
+\begin{lstlisting}[language=modelica]
+record experiment
+  Real StartTime(unit = "s") = 0;
+  Real StopTime(unit = "s");
+  Real Interval(unit = "s") = $\ldots{}$;
+  Real Tolerance = $\ldots{}$;
+end experiment;
+\end{lstlisting}
+The \lstinline!experiment! annotation defines the start time (\lstinline!StartTime!) in {[}s{]}, the stop time (\lstinline!StopTime!) in {[}s{]}, the suitable time resolution for the result grid (\lstinline!Interval!) in {[}s{]}, and the relative integration tolerance (\lstinline!Tolerance!) for simulation experiments to be carried out with the model or block at hand.
+Tools should use appropriate defaults for \lstinline!Interval! and \lstinline!Tolerance! instead of the $\ldots{}$.
 If \lstinline!StartTime! is not specified it is assumed to be \lstinline!0.0!.
+
+If \lstinline!StopTime! is set the model should be a simulation model.
```

Comment:
This logic is wrong. Better:
```suggestion
Providing a value for \lstinline!StopTime! is a necessary condition for being a simulation model, but the converse is not true as a partial model can have a \lstinline!StopTime! without being a simulation model, see \cref{FIXME}.
```

Reply:
This is also related to the question about why the use of _set_ was preferred over _modified_:
> In order to handle that in the future I ensure that the statement for StopTime requires that it is set; not modified.

Anyway, as suggested above, I think that _provide_ is an even better term to use in the context of annotations, as the use of _modified_ seems reading too much into the fact that annotations share grammar with the "real" modifications.

Reply:
I disagree that providing a value for StopTime is a necessary condition. We have 2 different use cases. One is a model that a user creates and wants to simulate. Another one is creating a library and indicating which models are examples that can be simulated as is. In the first use case, it has never been required to specify any experiment annotation at all, as far as I know. Usually, a tool would feel in any defaults. They may be different between the tools. But I do not know of a tool that would reject a model without StopTime. The second use case is basically how MSL treats examples: we consider it an example if there is an experiment annotation and StopTime is specified. So the requirement for the StopTime to be specified is necessary only if a library developer wants to indicate that a model or a block is a simulation model.

In a way, you can view it as any model that is simulated has an experiment annotations with all those fields and maybe some other tool-specific ones. Whatever user does not specify will be filled in by tool defaults. It's only to indicate which models in a library should be considered simulation models (for example, to indicate examples) that you need to specify experiment with StopTime. Without StopTime they would not be considered examples by default. That's my understanding of how experiment works.

Reply:
> But I do not know of a tool that would reject a model without StopTime.

I prefer to think of it as tools amending these models on the fly when the use requests a simulation. In System Modeler, the a dialog is presented where the user can modify or just confirm some tool specific defaults. For a smoother experience, there needs to be a `StopTime` in the model. From the specification's point of view, I don't think we need to consider these _simulation models_.

> The second use case is basically how MSL treats examples: we consider it an example if there is an experiment annotation and StopTime is specified. So the requirement for the StopTime to be specified is necessary only if a library developer wants to indicate that a model or a block is a simulation model.

Yes, and indicating that something is a simulation model comes with semantical rules, of particular importance when performing model validation (that is, not actually translating a model to be simulated). For example, when performing validation of a simulation model one should not give any diagnostics regarding the use of `Modelica.Utilities.Streams.print`.

Reply:
> > But I do not know of a tool that would reject a model without StopTime.
> 
> I prefer to think of it as tools amending these models on the fly when the use requests a simulation.

That's a non-working suggestion if the library is read-only.